list/PageableList.js

```diff
@@ -315,8 +221,87 @@ define([
 			};
 		}),
 
+		/*jshint maxcomplexity:14*/
+		computeProperties: function (props) {
+			if (this.pageLength > 0) {
+				if ("_busy" in props || "hideOnPageLoad" in props || "autoPaging" in props || "showNoItems" in props) {
+					this._updateListView();
+				}
+
+				if ("loadPreviousMessage" in props || "loadingMessage" in props || "_showPreviousPageLoader" in props) {
+					if (this.previousPageLoader && this.previousPageLoader.deliver) {
+						this.previousPageLoader.labels = {
+							loadMessage: string.substitute(this.loadPreviousMessage, this),
+							loadingMessage: this.loadingMessage
+						};
+						this.previousPageLoader.deliver();
+					}
+				}
+
+				if ("loadNextMessage" in props || "loadingMessage" in props || "_showNextPageLoader" in props) {
+					if (this.nextPageLoader && this.nextPageLoader.deliver) {
+						this.nextPageLoader.labels = {
+							loadMessage: string.substitute(this.loadNextMessage, this),
+							loadingMessage: this.loadingMessage
+						};
+						this.nextPageLoader.deliver();
+					}
+				}
+			}
+		},
+
 		//////////// Private methods ///////////////////////////////////////
 
+		/*
+		 * Ovveride deliteful/List#_updateListview to update the widget view.
+		 * @private
+		 */
+		_updateListView: function () {
+			this._displayedPanel = (this._busy && this.hideOnPageLoad && !this.autoPaging) ? "loading-panel" :
+				(this.containerNode && this.containerNode.children.length > 0) ?
+					"list" : ((this.showNoItems) ? "no-items" : "none");
+		},
+
+		/*
+		 * Handle click events on the widget.
+		 * If a loading is already in progress, this method
+		 * return undefined. In the other case, it starts a loading
+		 * and returns a Promise that resolves when the loading
+		 * has completed.
+		 * @returns {Promise} or null
+		 * @private
+		 */
+		_load: function (toLoad) {
+			if (this._dataLoaded && this._busy) { return; }
+			var loader = null;
+			// check if it's a synthetic event or not.
+			if (toLoad.target) {
+				loader = $(this._getLoader(toLoad.target)).hasClass("d-list-previous-loader") ? "prev" : "next";
+			} else {
+				loader = toLoad;
+			}
+			this._busy = true;
+			var self = this;
+			var f = (loader === "prev") ?  this._loadPreviousPage.bind(this) : this._loadNextPage.bind(this);
+			return f().then(function () {
+				self._busy = false;
+
+			});
+		},
+
+		/**
+		 * Returns the loader top node.
+		 * @param {HTML Element} node element
+		 * @return {HTML Element} the loader top node
+		 * @private
+		 */
+		_getLoader: function (node) {
+			while ("d-list-loader" !== node.tagName.toLowerCase()) {
+				node = node.parentElement;
+			}
+			return node;
+		},
```

Comment:
This seems like a lot of code to make _load() work regardless of whether the next or previous button was clicked.  How about having two separate methods _nextPageClickHandler() and _previousPageClickHandler()?